besu/src/main/java/org/hyperledger/besu/services/TransactionSimulationServiceImpl.java

```diff
@@ -51,50 +56,67 @@ public void init(final Blockchain blockchain, final TransactionSimulator transac
     this.transactionSimulator = transactionSimulator;
   }
 
+  @Override
+  public ProcessableBlockHeader simulatePendingBlockHeader() {
+    return transactionSimulator.simulatePendingBlockHeader();
+  }
+
   @Override
   public Optional<TransactionSimulationResult> simulate(
       final Transaction transaction,
       final Optional<StateOverrideMap> maybeStateOverrides,
-      final Optional<Hash> maybeBlockHash,
+      final Hash blockHash,
       final OperationTracer operationTracer,
       final boolean isAllowExceedingBalance) {
 
     final CallParameter callParameter = CallParameter.fromTransaction(transaction);
 
-    if (maybeBlockHash.isPresent()) {
-      final Hash blockHash = maybeBlockHash.get();
+    final var maybeBlockHeader =
+        blockchain.getBlockHeader(blockHash).or(() -> blockchain.getBlockHeaderSafe(blockHash));
```

Comment:
I think this can be just a call to getBlockHeaderSafe() since it defers to getBlockHeader if it is not in the cache

Reply:
but `getBlockHeaderSafe` is always synchronized, so we can a bit doing first the non synchronized way